server/athenian/api/controllers/search_controller/search_prs.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+import dataclasses
+from datetime import datetime, timedelta, timezone
+import logging
+from typing import Optional
+
+from aiohttp import web
+import aiomcache
+
+from athenian.api.async_utils import gather
+from athenian.api.db import DatabaseLike
+from athenian.api.internal.account import get_metadata_account_ids
+from athenian.api.internal.features.entries import PRFactsCalculator
+from athenian.api.internal.jira import get_jira_installation
+from athenian.api.internal.miners.filters import JIRAFilter, LabelFilter
+from athenian.api.internal.miners.github.bots import bots
+from athenian.api.internal.miners.github.pull_request import fetch_prs_numbers
+from athenian.api.internal.miners.types import (
+    JIRAEntityToFetch,
+    PRParticipants,
+    PRParticipationKind,
+    PullRequestFacts,
+)
+from athenian.api.internal.prefixer import Prefixer
+from athenian.api.internal.reposet import resolve_repos_with_request
+from athenian.api.internal.settings import LogicalRepositorySettings, ReleaseSettings, Settings
+from athenian.api.internal.with_ import resolve_withgroups
+from athenian.api.models.web import (
+    PullRequestDigest,
+    SearchPullRequestsRequest,
+    SearchPullRequestsResponse,
+)
+from athenian.api.request import AthenianWebRequest, model_from_body
+from athenian.api.response import model_response
+
+log = logging.getLogger(__name__)
+
+
+async def search_prs(request: AthenianWebRequest, body: dict) -> web.Response:
+    """Search pull requests that satisfy the query."""
+    search_request = model_from_body(SearchPullRequestsRequest, body)
+    connectors = _SearchPRsConnectors(request.mdb, request.pdb, request.rdb, request.cache)
+    account_info = await _build_account_info(search_request.account, request)
+    search_filter = await _build_filter(search_request, account_info, request)
+    pr_digests = await _search_pr_digests(search_filter, account_info, connectors)
+    return model_response(SearchPullRequestsResponse(pull_requests=pr_digests))
+
+
+async def _build_account_info(account: int, request: AthenianWebRequest) -> _SearchPRsAccountInfo:
+    meta_ids = await get_metadata_account_ids(account, request.sdb, request.cache)
+    prefixer = await Prefixer.load(meta_ids, request.mdb, request.cache)
+    settings = Settings.from_account(
+        account, prefixer, request.sdb, request.mdb, request.cache, request.app["slack"],
+    )
+    account_bots, release_settings, logical_settings = await gather(
+        bots(account, meta_ids, request.mdb, request.sdb, request.cache),
+        settings.list_release_matches(),
+        settings.list_logical_repositories(),
```

Comment:
We should ideally specify only repositories from the filter, if they exist (they will always exist in practice).

Reply:
not always, repo filter is optional in goals and so optional in the initial use case of this API call, anyway optimization done.